File: src/main/java/com/example/day08/leetcode_541/Solution.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        arr = list(s)
        for i in range(0,len(arr),k * 2):
            self.reverse(arr,i,min(i + k,len(arr)) - 1)
        return ''.join(arr)
    def reverse(self, arr, start, end):
        while start < end:
            arr[start], arr[end] = arr[end], arr[start]
            start += 1
            end -= 1

File: src/main/java/com/example/day08/leetcode_541/Solution.java
package com.example.day08.leetcode_541;

/**
 * 反转字符串2
 */
class Solution {
    public String reverseStr(String s, int k) {
        char[] arr = s.toCharArray();
        int n = arr.length;
        for(int i = 0;i < n;i += 2 * k){
            reverse(arr,i,Math.min(i + k,n) - 1);
        }
        return new String(arr);
    }

    private void reverse(char[] s,int l,int r){
        while(l < r){
            char tmp = s[l];
            s[l] = s[r];
            s[r] = tmp;
            l++;
            r --;
        }
    }

}